test/functional/ui/messages_spec.lua

```diff
@@ -795,6 +795,51 @@ describe('ui/ext_messages', function()
       pos = 9,
     }}}
   end)
+
+  it('hides prompt_for_number messages', function()
+    command('set spell')
+    feed('ihelllo<esc>')
+
+    feed('z=')
+    screen:expect{grid=[[
+      {r:helllo}                   |
+      {b:~                        }|
+      {b:~                        }|
+      {b:~                        }|
+      {b:^~                        }|
+    ]], messages={
+      {content = { { 'Change "helllo" to:\n 1 "Hello"\n 2 "Hallo"\n 3 "Helli"\nType number and <Enter> or click with mouse (empty cancels): ' } }, kind = ""}
+    }, attr_ids = {
+      r = {special = Screen.colors.Red, undercurl = true},
+      b = {foreground = Screen.colors.Blue, bold = true}
+    }}
+
+    feed('1')
+    screen:expect{grid=[[
+      {r:helllo}                   |
+      {b:~                        }|
+      {b:~                        }|
+      {b:~                        }|
+      {b:^~                        }|
+    ]], messages={
+      {content = { { 'Change "helllo" to:\n 1 "Hello"\n 2 "Hallo"\n 3 "Helli"\nType number and <Enter> or click with mouse (empty cancels): ' } }, kind = ""},
+      { content = { { "1" } }, kind = "" }
+    }, attr_ids = {
+      r = {special = Screen.colors.Red, undercurl = true},
+      b = {foreground = Screen.colors.Blue, bold = true}
+    }}
+
+    feed('<cr>')
+    screen:expect{grid=[[
+      ^Hello                    |
+      {1:~                        }|
+      {1:~                        }|
+      {1:~                        }|
+      {1:~                        }|
+    ]], messages={}}
```

Comment:
empty messages key can be removed (we always assert against unexpected ext_ taste)